### PR TITLE
Release 1.2.87

### PR DESCRIPTION
**Pull request by chubert-sb**
## MADiE PR

Jira Ticket: [MAT-8328](https://jira.cms.gov/browse/MAT-8328)
(Optional) Related Tickets:

### All Submissions
* [ ] This PR has the JIRA linked.
* [ ] Required tests are included.
* [ ] No extemporaneous files are included (i.e Complied files or testing results).
* [ ] This PR is merging into the **correct branch**.
* [ ] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).
* [ ] All CDN/Web dependencies are hosted internally (i.e MADiE-Root Repo).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

**Comment by chubert-sb** [comment]
Sample of what the custom error could look like. (From storybook)
Screenshot 2026-03-06 at 5 02 03 PM

**Comment by RohitKandimalla** [comment]
Missing some of the ACs.
We need this message on the dialog "! This action cannot be undone!" Is it handled ? I am asking solely based on the screenshot you added to the PR.

File: react/components/MadieDeleteDialog/index.jsx [hunk 1]
```diff
@@ -51,7 +62,7 @@ const MadieDeleteDialog = ({
                             options
                         )}
                     </span>
-                    ?
+                    {otherDialogProps?.statement ? "" : "?"}
```
Comment (RohitKandimalla) [inline react/components/MadieDeleteDialog/index.jsx:65]: I don't think this is helping us at least for this requirement

File: react/components/MadieDeleteDialog/MadieDeleteDialog.stories.js [hunk 2]
```diff
@@ -113,3 +113,32 @@ export const DeleteDialogWithAlternateText = () => {
...
+                onClose={onClose}
+                hideWarning={true}
+                dialogTitle="Are you sure?"
+                customDialog={
+                    "You are choosing to delete the following Test Case(s)!"
+                }
+                alternateText="Delete Case(s)"
+                statement={true}
+                name={
+                    "<p><ol><li><p><strong>test2</strong></p></li><li><p><strong>test3</strong></p></li></ol></p>"
```
Comment (RohitKandimalla) [inline react/components/MadieDeleteDialog/MadieDeleteDialog.stories.js:139]: The attribute "name" throws me off. The same goes with AlternateText. The same goes with customDialog - it it technically not a dialog, it is dialogBody.

**Comment by RohitKandimalla** [comment]
> Missing some of the ACs. We need this message on the dialog "! This action cannot be undone!" Is it handled ? I am asking solely based on the screenshot you added to the PR.

Nvm, we have to send an additional parameter for it to be displayed.

File: react/components/MadieDeleteDialog/index.jsx (hunk 1 above)
Comment (chubert-sb) [reply react/components/MadieDeleteDialog/index.jsx:65]: Most of our delete dialogs are questions. This is the only one so far that isn't. It just helps hide the question mark.

File: react/components/MadieDeleteDialog/MadieDeleteDialog.stories.js (hunk 2 above)
Comment (chubert-sb) [reply react/components/MadieDeleteDialog/MadieDeleteDialog.stories.js:139]: Yeah I don't disagree with you on some of these naming choices. I changed it to customDialogBody. The other attributes would require many more updates I think.

**Review by RohitKandimalla — APPROVED** [review]
